dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -713,11 +713,11 @@
 			<!-- TODO: these steps should probably run after mark. -->
 			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeStep="MarkStep" Condition="'$(_AreAnyAssembliesTrimmed)' == 'true'" Type="Xamarin.Linker.Steps.PreMarkDispatcher" />
 			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeStep="MarkStep" Type="Xamarin.Linker.ManagedRegistrarStep" Condition="'$(Registrar)' == 'managed-static'" />
+			<_TrimmerCustomSteps Include="$(_AdditionalTaskAssembly)" BeforeSteps="MarkStep" Condition="'$(_AreAnyAssembliesTrimmed)' == 'true'" Type="Xamarin.Linker.Steps.PreserveBlockCodeHandler" />
```

Comment:
I'm not sure I understand this change, the only difference is the addition of `BeforeSteps="MarkStep"`, it'll still use the same PreserveBlockCodeHandler implementation, not the new one you added.

It seems to be this line should be entirely removed, and then a new MSBuild target that runs before ILLink is needed, which runs the pre-trimmer steps.